scripts/apps/search/components/fields/translations.tsx

```diff
@@ -38,41 +52,47 @@ export class Translations extends React.PureComponent<IProps> {
             );
         }
 
+        if (this.state.popup != null) {
+            output.push(
+                <ItemListPopup
+                    key="popup"
+                    target={this.state.popup.target}
+                    label={this.state.popup.label}
+                    close={this.closePopup}>
+                    <TranslationsList
+                        ids={this.state.popup.ids}
+                        onClick={(item) => {
+                            ng.get('$rootScope').$broadcast('broadcast:preview', {item});
+                        }}
+                    />
+                </ItemListPopup>,
+            );
+        }
+
         return output.length > 0 ? output : null;
     }
 
     renderOriginalArticle(elem: EventTarget) {
-        this.renderPopup(
-            gettext('Original Article'),
-            [this.props.item.translated_from],
-            elem,
-        );
+        this.setState({
```

Comment:
Looks like you are going to end up state being out-of-sync from UI. When you click the close button, the DOM will be imperatively modified while this state will still contain the data of now closed popup.

If you were to go with the popup interface I suggested, that wouldn't happen. Why did you decide against it? Do you think it was a bad approach or are you trying to save time?

Reply:
the dom will be only modified after the state changes

Reply:
Right, I've mistaken how the data flows. Can you address the second question? :slightly_smiling_face:

Reply:
I didn't see the need to complicate it, so instead of using a function it just renders the children like any other component.

Reply:
I hear you, but your version isn't without "complications" either. You now have to store ids,label and dom node in the state as well as have `renderOriginalArticle`, `renderTranslations` and `closePopup` methods in the component, which you wouldn't have needed otherwise.